src/responder/store.ts

```diff
@@ -0,0 +1,115 @@
+import { StartStopService } from "../dataEntities";
+import { GasQueueItem, GasQueue } from "./gasQueue";
+import { LevelUp } from "levelup";
+import EncodingDown from "encoding-down";
+const sub = require("subleveldown");
+
+export class ResponderStore extends StartStopService {
+    public get transactions(): ReadonlyMap<string, GasQueueItem> {
+        return this.mTransactions;
+    }
+    private readonly mTransactions: Map<string, GasQueueItem> = new Map();
+    /**
+     * A single global lock on this store to be taken out whenever reading or
+     * writing to the store.
+     */
+    public readonly lock: string;
+    private mQueue: GasQueue;
+    public get queue() {
+        return this.mQueue;
+    }
+
+    private readonly subDb: LevelUp<EncodingDown<string, any>>;
+    private readonly queueKey: string;
+    /**
+     * A persistent store for responder data.
+     * @param db A backend database for the store
+     * @param responderAddress The address of the responder using this store. Responder public keys can only
+     * be used by one responder at a time.
+     */
+    constructor(db: LevelUp<EncodingDown<string, any>>, responderAddress: string, seedQueue: GasQueue) {
+        //TODO:261: this encoding is actually a requirement
```

Comment:
TODO yet to be addressed?

Reply:
No, it's been addressed. The sub db needs to explicitly insist on json encoding, rather than relying on the encoding of the passed in db